### DISCUSSION TITLE
useRecoilCallback with set vs useSetRecoilState best practice

### DISCUSSION THREAD

**Discussion by Dattaya** [facebookexperimental/Recoil#2050] [Q&A] [answered]
A similar question was asked previosly (https://github.com/facebookexperimental/Recoil/discussions/1193), but for some reason a selector was used in the example, so it's still not clear which one is preferable or it's just a matter of taste. I don't think there's any difference,  but it would be nice to know for sure. 
```jsx
const setSomething = useSetRecoilState(myAtom);
const handleOnClick = useCallback((event) => {
  // ...
  setSomething('something')
}, [setSomething])
```
vs
```jsx
const handleOnClick = useRecoilCallback(({ set }) => (event) => {
  // ...
  set(myAtom, 'something')
}, [])
```

**Comment by drarmstr** [comment, marked as answer]
`useSetRecoilState()` is more efficient, but `useRecoilCallback()` can handle more complex situations.

**Reply by Dattaya** [reply]
got it, thanks so much for the explanation
